### DISCUSSION TITLE
Does Open Watcom support 64-bit pointers

### DISCUSSION THREAD

**Discussion by CyberSimian** [open-watcom/open-watcom-v2#739] [Q&A] [answered]
I have downloaded and installed the build from 2021-09-01, but I am unable to decide whether OW 2.0 supports 64-bit pointers. 

I set the PATH environment variable to include the BINNT64 and BINNT folders (in that order), and I specified a build target of "/bt=NT64", but when I compile my test program that displays the number of bytes in a "char*" I always get "4".  The best that I have been able to do is add "far" to the declaration, and then the pointer size becomes "6".  

Can OW generate programs that use 64-bit pointers, or is this still in development? If it is possible, what do I need to specify the get 64-bit pointers in my application? Thank you.

-- from CyberSimian in the UK

**Comment by jmalak** [comment, marked as answer]
No, OW is still generating 16 or 32 bit code only even if all tools were ported to 64-bit platforms ( Windows and Linux ).
It will be next step to enable OW generate 64-bit code.

**Reply by CyberSimian** [reply]
Thank you for the info. 
I look forward to the day when OW can generate 64-bit pointers.
Thank you for the work that you have done so far.

-- from CyberSimian in the UK
